processor/transaction/processor.go

```diff
@@ -20,22 +19,17 @@ const (
 
 func NewProcessor() pr.Processor {
 	return &processor{
-		payload: &payload{},
-		schema:  pr.CreateSchema(transactionSchema, processorName),
+		schema: pr.CreateSchema(transactionSchema, processorName),
 	}
 }
 
 type processor struct {
-	payload *payload
-	schema  *jsonschema.Schema
+	schema *jsonschema.Schema
 }
 
-func (p *processor) Transform() []beat.Event {
-	return p.payload.transform()
-}
-
-func (p *processor) Validate(reader io.Reader) error {
-	return pr.Validate(reader, p.schema, p.payload)
+func (p *processor) Validate(reader io.Reader) (pr.Payload, error) {
+	var payload Payload
+	return &payload, pr.Validate(reader, p.schema, &payload)
```

Comment:
@simitt If I were to add few points here,I like what you said and returning payload sounds better.

As we are registering the processors only once using NewProcessor,they are singleton.So,ideally our processor methods which act on input request should be stateless and and not mutate data(payload) but instead work like functions.